core/src/main/scala/cats/TraverseFilter.scala

```diff
@@ -124,7 +124,12 @@ trait TraverseFilter[F[_]] extends FunctorFilter[F] {
 
   /**
    * Removes duplicate elements from a list, keeping only the first occurrence.
-   * This is usually faster than ordDistinct, especially for things that have a slow comparion (like String).
+   * This is usually faster than ordDistinct, especially for things that have a slow comparison (like String).
+   * 
+   * @note the passed `Hash` typeclass is not currently in use.
+   * @todo replace `HashSet` from Scala library with one that can make use of the `Hash` typeclass
+   *       instead of the Java's `hashCode` method. Consider a candidate implementation in
+   *       [[https://github.com/typelevel/cats/pull/4185 PR#4185]].
    */
   def hashDistinct[A](fa: F[A])(implicit H: Hash[A]): F[A] =
     traverseFilter[State[HashSet[A], *], A, A](fa)(a =>
```

Comment:
Raises the "parameter `H: Hash[A]` not used" warning because it is really so.

I'd prefer to keep this warning in place until Cats receive a suitable `HashSet` implementation (perhaps by merging #4185).